cmake/Modules/FindMETIS.cmake

```diff
@@ -45,16 +45,24 @@ if (METIS_INCLUDE_DIR)
     string(REGEX REPLACE "^#define[\t ]+METIS_VER_MINOR[\t ]+([0-9]+).*" "\\1" METIS_VERSION_MINOR "${metis_version_str_minor}")
     set(METIS_VERSION ${METIS_VERSION_MAJOR}.${METIS_VERSION_MINOR})
     find_library(METIS_LIBRARY ${METIS_LIB_NAME} HINTS ${METIS_DIR} ENV METIS_DIR PATH_SUFFIXES lib lib64)
+    if (METIS_VERSION VERSION_GREATER_EQUAL 5.0)
+        find_library(METIS_GKLIB_LIBRARY GKlib HINTS ${METIS_DIR} ENV METIS_DIR PATH_SUFFIXES lib lib64)
+    else()
+        # set it to a non-empty value to satisfy find_package_handle_standard_args
+        set(METIS_GKLIB_LIBRARY libGKlib.so)
+    endif()
 endif()
 
 include(FindPackageHandleStandardArgs)
-find_package_handle_standard_args(METIS REQUIRED_VARS METIS_LIBRARY METIS_INCLUDE_DIR VERSION_VAR METIS_VERSION)
+find_package_handle_standard_args(METIS REQUIRED_VARS METIS_LIBRARY METIS_GKLIB_LIBRARY METIS_INCLUDE_DIR VERSION_VAR METIS_VERSION)
```

Comment:
`find_package_handle_standard_args` only checks whether it is valid. what is the valid meaning here? they mention valid path. but is it just regex check or the existence check? This changes will work when they only do regex check. Otherwise, they will always be failed when gklib is not existed (i.e. version < 5)

Reply:
valid in this case means that it would evaluate to true in an `if` statement, see https://github.com/Kitware/CMake/blob/aefd952085b54089e20b51fabf2ebf926e886fa9/Modules/FindPackageHandleStandardArgs.cmake#L488, because `SOMETHING-NOTFOUND` and a handful of other strings are evaluated to false, which is how this works normally.

Reply:
I see. Thanks for explanations!